app/autoload.php

```diff
@@ -5,36 +5,48 @@
  * Copyright © Magento, Inc. All rights reserved.
  * See COPYING.txt for license details.
  */
+declare(strict_types=1);
+
 use Magento\Framework\Autoload\AutoloaderRegistry;
 use Magento\Framework\Autoload\ClassLoaderWrapper;
 
 /**
  * Shortcut constant for the root directory
  */
-define('BP', dirname(__DIR__));
+\define('BP', \dirname(__DIR__));
 
-define('VENDOR_PATH', BP . '/app/etc/vendor_path.php');
+\define('VENDOR_PATH', BP . '/app/etc/vendor_path.php');
 
-if (!file_exists(VENDOR_PATH)) {
+if (!\is_readable(VENDOR_PATH)) {
     throw new \Exception(
         'We can\'t read some files that are required to run the Magento application. '
          . 'This usually means file permissions are set incorrectly.'
     );
 }
 
-$vendorDir = require VENDOR_PATH;
-$vendorAutoload = BP . "/{$vendorDir}/autoload.php";
+$vendorAutoload = (
+    static function (): ?string {
+        $vendorDir = require VENDOR_PATH;
+
```

Comment:
Hi @rvitaliy. Thank you for your collaboration. Could I kindly ask you to provide a comment for these two declarations of `$vendorAutoload` with highlighting what is a particular block for? At first glance, these blocks of code look very similar (almost the same) and it's hard to guess why we need such an approach. I see that in first case we have the trailing slash, but let's make developers life a little bit easier and highlight this case in the comments. Thank you!

Reply:
@rogyar 
sorry, but i can't do it.
personally i don't undestand why we need the second check without BP prefix.
i'm refactor the old code without change to the logics, make it more strict and readable.
IMHO: we can remove the second block of check, `app/{$vendor}` is not a valid path for vendor.

Reply:
Good point. I see that this functionality has been recently added. So I assume it was done for a purpose.

https://github.com/magento/magento2/commit/2054fab1731a63728a6b4e573cefbbf3e64d6c85

Unfortunately, there's no additional information about this change. Well then, let's leave it as it is.

Reply:
@rogyar, that commit is from https://github.com/magento/magento2/pull/18849, so there is your missing information 🙂

Reply:
so the second check is need to autoload vendors from the global path